tests/devices/test_default_mixed_jax.py

```diff
@@ -0,0 +1,635 @@
+                (1 + np.cos(x) * np.cos(y)) / 2,
+                (1 - np.cos(x) * np.cos(y)) / 2,
+            ]
+        )
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+        res = jax.jacobian(circuit, (0, 1))(x, y)
+        expected = np.array(
+            [
+                [-np.sin(x), -np.sin(x) * np.cos(y) / 2, np.cos(y) * np.sin(x) / 2],
+                [0, -np.cos(x) * np.sin(y) / 2, np.cos(x) * np.sin(y) / 2],
+            ]
+        )
+        assert np.allclose(res, expected, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("jacobian_fn", [jax.jacfwd, jax.jacrev])
+    @pytest.mark.parametrize("decorator", decorators)
+    def test_batching(self, jacobian_fn, decorator, tol):
+        """Tests that the gradient of the qnode is correct with batching"""
+        dev = qml.device("default.mixed", wires=2)
+
+        if decorator == jax.jit:
+            pytest.skip("Batching currently not supported for JAX jit")
+
+        @qml.batch_params
+        @qml.qnode(dev, diff_method="backprop", interface="jax")
+        def circuit(a, b):
+            qml.RX(a, wires=0)
+            qml.CRX(b, wires=[0, 1])
+            return qml.expval(qml.PauliZ(0) @ qml.PauliZ(1))
+
+        a = jnp.array([-0.234, 0.678])
+        b = jnp.array([0.654, 1.236])
+
+        res = decorator(circuit)(a, b)
+        expected_cost = 0.5 * (np.cos(a) * np.cos(b) + np.cos(a) - np.cos(b) + 1)
+        assert np.allclose(res, expected_cost, atol=tol, rtol=0)
+
+        res_a, res_b = decorator(jacobian_fn(circuit, (0, 1)))(a, b)
+        expected_a, expected_b = [
+            -0.5 * np.sin(a) * (np.cos(b) + 1),
+            0.5 * np.sin(b) * (1 - np.cos(a)),
+        ]
+
+        assert np.allclose(jnp.diag(res_a), expected_a, atol=tol, rtol=0)
+        assert np.allclose(jnp.diag(res_b), expected_b, atol=tol, rtol=0)
+
+
+class TestHighLevelIntegration:
+    """Tests for integration with higher level components of PennyLane."""
+
+    def test_template_integration(self):
+        """Test that a PassthruQNode default.mixed with JAX works with templates."""
+        dev = qml.device("default.mixed", wires=2)
+
+        @qml.qnode(dev, interface="jax", diff_method="backprop")
+        def circuit(weights):
+            qml.templates.StronglyEntanglingLayers(weights, wires=[0, 1])
+            return qml.expval(qml.PauliZ(0))
+
+        shape = qml.templates.StronglyEntanglingLayers.shape(n_layers=2, n_wires=2)
+        weights = jnp.array(np.random.random(shape))
+
+        grad = jax.grad(circuit)(weights)
+        assert grad.shape == weights.shape
+
+    def test_qnode_collection_integration(self):
+        """Test that a PassthruQNode default.mixed with JAX works with QNodeCollections."""
+        dev = qml.device("default.mixed", wires=2)
+
+        obs_list = [qml.PauliX(0) @ qml.PauliY(1), qml.PauliZ(0), qml.PauliZ(0) @ qml.PauliZ(1)]
+        qnodes = qml.map(qml.templates.StronglyEntanglingLayers, obs_list, dev, interface="jax")
```

Comment:
Maybe worth commenting that we're depending on the fact that backprop is the best diff method and set implicitly here.

Reply:
Added an explicit `diff_method="backprop"` flag to the `qml.map` call. Hopefully this works.